code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -354,3 +354,27 @@
 /mob/living/simple_animal/drone/flash_eyes(intensity = 1, override_blindness_check = 0, affect_silicon = 0, visual = 0)
 	if(affect_silicon)
 		return ..()
+
+/mob/living/silicon/robot/drone/decompile_act(obj/item/matter_decompiler/C, mob/user)
+	if(!client && istype(user, /mob/living/silicon/robot/drone))
+		var/mob/living/silicon/robot/drone/D = C.loc
+		to_chat(D, "<span class='warning'>You begin decompiling the other drone.</span>")
+
+		if(!do_after(D, 50, target = src.loc))
+			to_chat(D, "<span class='warning'>You need to remain still while decompiling such a large object.</span>")
+			return
+
+		if(!src || !D) return ..()
+
+		to_chat(D, "<span class='warning'>You carefully and thoroughly decompile your downed fellow, storing as much of its resources as you can within yourself.</span>")
+
+		new/obj/effect/decal/cleanable/blood/oil(get_turf(src))
+		qdel(src)
+
+		C.stored_comms["metal"] += 15
```

Comment:
Best to put these lines from 374 - 379 above the `qdel(src)` call. In case the SSGarbage queue is full and it will hard dell the item.
Feel free to ping me if you need some more explanation.

Reply:
Nope, makes sense.

Reply:
What do you mean, if the queue is full?

Reply:
If I read it right the garbage collection will hard del items if its queue is full when ypu call qdel. In that case the proc woll instantly stop since src is gone.

Reply:
Yeah, that's not what that means.

There's multiple queues in the GC--the first one is the regular queue , the second one is the hard queue.  This is a safety for if someone tries to somehow add something to a third level queue which doesn't exist---which would throw really bad runtimes and cause issues for GC'ing overall....so in thist case, given how important garbage collection is--better to just go "welp, someone did something really dumb; better hard delete it....".

Reply:
Ah okay got it now yeah. I mistook the level with something else.
So only time it can instantly delete is when the object itself returns a hint saying it wants to be hard deleted. Which is not the case here.
Thanks for the info! The code should function fine then in these areas.